[zlib] warning: Broken installation of vcpkg port zlib (Android)

**Describe the bug**
Is there a reason why I get "Broken installation of vcpkg port zlib" after updating my vcpkg ports.

**Environment**
- Windows + NDK targeting Android 29 on ARM64
<details>
 <summary>Full gory details</summary>

```
    -- Custom Android NDK Toolchain
    -- root:    C:/Microsoft/AndroidSDK/25
    -- ndk:     C:/Microsoft/AndroidSDK/25/ndk
    -- ndk_ver: 22.1.7171670
    -- sys_ver: 29
    -- ninja:   C:/Microsoft/AndroidSDK/25/cmake/3.10.2.4988404/bin/ninja.exe
    -- Android: Targeting API '21' with architecture 'arm64', ABI 'arm64-v8a', and processor 'aarch64'
    -- Custom Android NDK Toolchain
    -- root:    C:/Microsoft/AndroidSDK/25
    -- ndk:     C:/Microsoft/AndroidSDK/25/ndk
    -- ndk_ver: 22.1.7171670
    -- sys_ver: 29
    -- ninja:   C:/Microsoft/AndroidSDK/25/cmake/3.10.2.4988404/bin/ninja.exe
    -- Android: Selected unified Clang toolchain
    -- The C compiler identification is Clang 11.0.5
    -- Detecting C compiler ABI info
    -- Detecting C compiler ABI info - done
    -- Check for working C compiler: C:/Microsoft/AndroidSDK/25/ndk/22.1.7171670/toolchains/llvm/prebuilt/windows-x86_64/bin/clang.exe - skipped
    -- Detecting C compile features
    -- Detecting C compile features - done
    -- The CXX compiler identification is Clang 11.0.5
    -- Detecting CXX compiler ABI info
    -- Detecting CXX compiler ABI info - done
    -- Check for working CXX compiler: C:/Microsoft/AndroidSDK/25/ndk/22.1.7171670/toolchains/llvm/prebuilt/windows-x86_64/bin/clang++.exe - skipped
    -- Detecting CXX compile features
    -- Detecting CXX compile features - done
    #COMPILER_HASH#4cd580d23fc13bc8374d090d9f853424e245f5ad
    #COMPILER_C_HASH#363c3ee7d32c9d699f2db05451e67a8b4b09faa9
    #COMPILER_C_VERSION#11.0.5
    #COMPILER_C_ID#Clang
    #COMPILER_CXX_HASH#363c3ee7d32c9d699f2db05451e67a8b4b09faa9
    #COMPILER_CXX_VERSION#11.0.5
    #COMPILER_CXX_ID#Clang
```

</details>

**To Reproduce**
Steps to reproduce the behavior:
1. ./vcpkg install zlib
2. Try to build anythging statically that uses zlib on Android.

**Expected behavior**
No warnings.

**Failure logs** 

<details>
  <summary>Bloody logs!</summary>

```cmake
    -- OpenCV disables pkg-config to avoid using of host libraries. Consider using PKG_CONFIG_LIBDIR to specify target SYSROOT
    CMake Warning at C:/extern/workbench/test/build/outputs/opencv/ndk-android-aarch64-debug/vcpkg_installed/ndk-android-aarch64/share/zlib/vcpkg-cmake-wrapper.cmake:6 (message):
    ZLIB_ROOT:
    C:/extern/workbench/test/build/outputs/opencv/ndk-android-aarch64-debug/vcpkg_installed/ndk-android-aarch64
    Call Stack (most recent call first):
    C:/extern/workbench/test-app/thirdparty/vcpkg/scripts/buildsystems/vcpkg.cmake:738 (include)
    cmake/OpenCVFindLibsGrfmt.cmake:14 (find_package)
    CMakeLists.txt:710 (include)
    CMake Warning at C:/extern/workbench/test/build/outputs/opencv/ndk-android-aarch64-debug/vcpkg_installed/ndk-android-aarch64/share/zlib/vcpkg-cmake-wrapper.cmake:7 (message):
    ZLIB_INCLUDE_DIR:
    C:/extern/workbench/test/build/outputs/opencv/ndk-android-aarch64-debug/vcpkg_installed/ndk-android-aarch64/include
    Call Stack (most recent call first):
    C:/extern/workbench/test-app/thirdparty/vcpkg/scripts/buildsystems/vcpkg.cmake:738 (include)
    cmake/OpenCVFindLibsGrfmt.cmake:14 (find_package)
    CMakeLists.txt:710 (include)
    CMake Warning at C:/extern/workbench/test/build/outputs/opencv/ndk-android-aarch64-debug/vcpkg_installed/ndk-android-aarch64/share/zlib/vcpkg-cmake-wrapper.cmake:8 (message):
    ZLIB_LIBRARY_RELEASE:
    C:/Microsoft/AndroidSDK/25/ndk/22.1.7171670/toolchains/llvm/prebuilt/windows-x86_64/sysroot/usr/lib/aarch64-linux-android/29/libz.so
    Call Stack (most recent call first):
    C:/extern/workbench/test-app/thirdparty/vcpkg/scripts/buildsystems/vcpkg.cmake:738 (include)
    cmake/OpenCVFindLibsGrfmt.cmake:14 (find_package)
    CMakeLists.txt:710 (include)
    CMake Warning at C:/extern/workbench/test/build/outputs/opencv/ndk-android-aarch64-debug/vcpkg_installed/ndk-android-aarch64/share/zlib/vcpkg-cmake-wrapper.cmake:9 (message):
    ZLIB_LIBRARY_DEBUG: ZLIB_LIBRARY_DEBUG-NOTFOUND
    Call Stack (most recent call first):
    C:/extern/workbench/test-app/thirdparty/vcpkg/scripts/buildsystems/vcpkg.cmake:738 (include)
    cmake/OpenCVFindLibsGrfmt.cmake:14 (find_package)
    CMakeLists.txt:710 (include)
    CMake Warning at C:/extern/workbench/test/build/outputs/opencv/ndk-android-aarch64-debug/vcpkg_installed/ndk-android-aarch64/share/zlib/vcpkg-cmake-wrapper.cmake:10 (message):
    Broken installation of vcpkg port zlib
    Call Stack (most recent call first):
    C:/extern/workbench/test-app/thirdparty/vcpkg/scripts/buildsystems/vcpkg.cmake:738 (include)
    cmake/OpenCVFindLibsGrfmt.cmake:14 (find_package)
    CMakeLists.txt:710 (include)
    -- Could NOT find PkgConfig (missing: PKG_CONFIG_EXECUTABLE)
    -- Consider adding OPENCV_ALLOCATOR_STATS_COUNTER_TYPE=int/int64_t according to your build configuration
    -- Excluding from source files list: modules/imgproc/src/corner.avx.cpp
    -- Excluding from source files list: modules/imgproc/src/imgwarp.avx2.cpp
    -- Excluding from source files list: modules/imgproc/src/imgwarp.sse4_1.cpp
```

</details>




**Additional context**
The bisect change that was made, I guess there's a good reason for that warning.

https://github.com/microsoft/vcpkg/blob/7dbc05515b44bf54d2a42b4da9d1e1f910868b86/ports/zlib/portfile.cmake#L34

https://github.com/microsoft/vcpkg/blob/7dbc05515b44bf54d2a42b4da9d1e1f910868b86/ports/zlib/vcpkg-cmake-wrapper.cmake#L6

I was able to get a local repro of this issue pretty easily:

```
> $Env:ANDROID_NDK_HOME = "$HOME\AppData\Local\Android\Sdk\ndk\22.1.7171670\"
> .\vcpkg.exe install opencv --triplet=x64-android

<snip>

-- Configuring x64-android
CMake Error at scripts/cmake/vcpkg_execute_required_process.cmake:128 (message):
    Command failed: ninja -v
    Working Directory: D:/Source/vcpkg/buildtrees/opencv4/x64-android-rel/vcpkg-parallel-configure
    Error code: 1
    See logs for more information:
      D:\Source\vcpkg\buildtrees\opencv4\config-x64-android-out.log

Call Stack (most recent call first):
  installed/x64-windows/share/vcpkg-cmake/vcpkg_cmake_configure.cmake:415 (vcpkg_execute_required_process)
  ports/opencv4/portfile.cmake:390 (vcpkg_cmake_configure)
  scripts/ports.cmake:145 (include)

<snip>
```

Relevant error message from the configure log:

```
CMake Error at D:/Source/vcpkg/installed/x64-android/share/zlib/vcpkg-cmake-wrapper.cmake:5 (message):
  Broken installation of vcpkg port zlib
Call Stack (most recent call first):
  D:/Source/vcpkg/scripts/buildsystems/vcpkg.cmake:743 (include)
  cmake/OpenCVFindLibsGrfmt.cmake:14 (find_package)
  CMakeLists.txt:724 (include)
```

I was able to successfully build by simply removing the `if(ANDROID)` hacks from _OpenCVFindLibsGrfmt.cmake_. Opening a PR now...

--- a/ports/opencv3/0012-fix-zlib.patch
+++ b/ports/opencv3/0012-fix-zlib.patch
@@ -1,0 +1,14 @@
+diff --git a/cmake/OpenCVFindLibsGrfmt.cmake b/cmake/OpenCVFindLibsGrfmt.cmake
+index 6dbeca9..08e2f4a 100644
+--- a/cmake/OpenCVFindLibsGrfmt.cmake
++++ b/cmake/OpenCVFindLibsGrfmt.cmake
+@@ -5,6 +5,9 @@
+ # --- zlib (required) ---
+ if(BUILD_ZLIB)
+   ocv_clear_vars(ZLIB_FOUND)
++elseif(1)
++  ocv_clear_internal_cache_vars(ZLIB_LIBRARY ZLIB_INCLUDE_DIR)
++  find_package(ZLIB "${MIN_VER_ZLIB}" REQUIRED)
+ else()
+   ocv_clear_internal_cache_vars(ZLIB_LIBRARY ZLIB_INCLUDE_DIR)
+   if(ANDROID)

--- a/ports/opencv3/portfile.cmake
+++ b/ports/opencv3/portfile.cmake
@@ -26,6 +26,7 @@ vcpkg_from_github(
       0009-fix-protobuf.patch
       0010-fix-uwp-tiff-imgcodecs.patch
       0011-remove-python2.patch
+      0012-fix-zlib.patch
 )
 
 if(VCPKG_TARGET_ARCHITECTURE STREQUAL "arm64")

--- a/ports/opencv3/vcpkg.json
+++ b/ports/opencv3/vcpkg.json
@@ -1,7 +1,7 @@
 {
   "name": "opencv3",
   "version": "3.4.16",
-  "port-version": 2,
+  "port-version": 3,
   "description": "computer vision library",
   "homepage": "https://github.com/opencv/opencv",
   "license": "BSD-3-Clause",

--- a/ports/opencv4/0012-fix-zlib.patch
+++ b/ports/opencv4/0012-fix-zlib.patch
@@ -1,0 +1,14 @@
+diff --git a/cmake/OpenCVFindLibsGrfmt.cmake b/cmake/OpenCVFindLibsGrfmt.cmake
+index 8ea5497..7ed3f6b 100644
+--- a/cmake/OpenCVFindLibsGrfmt.cmake
++++ b/cmake/OpenCVFindLibsGrfmt.cmake
+@@ -5,6 +5,9 @@
+ # --- zlib (required) ---
+ if(BUILD_ZLIB)
+   ocv_clear_vars(ZLIB_FOUND)
++elseif(1)
++  ocv_clear_internal_cache_vars(ZLIB_LIBRARY ZLIB_INCLUDE_DIR)
++  find_package(ZLIB "${MIN_VER_ZLIB}" REQUIRED)
+ else()
+   ocv_clear_internal_cache_vars(ZLIB_LIBRARY ZLIB_INCLUDE_DIR)
+   if(ANDROID)

--- a/ports/opencv4/portfile.cmake
+++ b/ports/opencv4/portfile.cmake
@@ -26,6 +26,7 @@ vcpkg_from_github(
       0009-fix-protobuf.patch
       0010-fix-uwp-tiff-imgcodecs.patch
       0011-remove-python2.patch
+      0012-fix-zlib.patch
 )
 
 if(VCPKG_TARGET_ARCHITECTURE STREQUAL "arm64")

--- a/ports/opencv4/vcpkg.json
+++ b/ports/opencv4/vcpkg.json
@@ -1,7 +1,7 @@
 {
   "name": "opencv4",
   "version": "4.5.4",
-  "port-version": 3,
+  "port-version": 4,
   "description": "computer vision library",
   "homepage": "https://github.com/opencv/opencv",
   "dependencies": [

--- a/versions/baseline.json
+++ b/versions/baseline.json
@@ -5002,11 +5002,11 @@
     },
     "opencv3": {
       "baseline": "3.4.16",
-      "port-version": 2
+      "port-version": 3
     },
     "opencv4": {
       "baseline": "4.5.4",
-      "port-version": 3
+      "port-version": 4
     },
     "opendnp3": {
       "baseline": "3.1.1",

--- a/versions/o-/opencv3.json
+++ b/versions/o-/opencv3.json
@@ -1,6 +1,11 @@
 {
   "versions": [
     {
+      "git-tree": "76d605b750c5535590489574dc24fa30ffe249f8",
+      "version": "3.4.16",
+      "port-version": 3
+    },
+    {
       "git-tree": "a11dd583ecb7212732fedb0ab3e3308ea84f10d9",
       "version": "3.4.16",
       "port-version": 2

--- a/versions/o-/opencv4.json
+++ b/versions/o-/opencv4.json
@@ -1,6 +1,11 @@
 {
   "versions": [
     {
+      "git-tree": "ce474a40e6572ced8dd1229f22f6cbed9cc74212",
+      "version": "4.5.4",
+      "port-version": 4
+    },
+    {
       "git-tree": "b170a087d34521462cd247a24f230627a272975e",
       "version": "4.5.4",
       "port-version": 3

